initial_setup/__init__.py

```diff
@@ -275,25 +290,43 @@ def _apply(self):
             log.debug("executing %s", section_msg)
             section.execute()
 
-        # Prepare the user database tools
-        u = Users()
-
-        sections = [self.data.group, self.data.user, self.data.rootpw]
-
+        # Configure groups, users & root account
+        #
+        # NOTE: We only configure groups, users & root account if the respective
+        #       kickstart commands are *not* seen in the input kickstart.
+        #       This basically means that we will configure only what was
+        #       set in the Initial Setup UI and will not attempt to configure
+        #       anything that looks like it was configured previously in
+        #       the Anaconda UI or installation kickstart.
         users_proxy = USERS.get_proxy()
-        self.data.rootpw.seen = users_proxy.IsRootpwKickstarted
 
-        for section in sections:
-            section_msg = "%s on line %d" % (repr(section), section.lineno)
-            if section.seen:
-                log.debug("skipping %s", section_msg)
-                continue
-            log.debug("executing %s", section_msg)
-            section.execute(storage=None, ksdata=self.data, users=u)
+        groups_task = users_proxy.ConfigureGroupsWithTask((util.getSysroot()))
+        if self._groups_already_configured and not reconfig_mode:
+            log.debug("skipping user group configuration - already configured")
+        elif users_proxy.Groups:  # only run of there are some groups to create
+            task_proxy = USERS.get_proxy(groups_task)
+            log.debug("configuring user groups via %s task", task_proxy.Name)
+            sync_run_task(task_proxy)
+
+        users_task = users_proxy.ConfigureUsersWithTask((util.getSysroot()))
+        if self._users_already_configured and not reconfig_mode:
+            log.debug("skipping user configuration - already configured")
+        elif users_proxy.Users:  # only run if there are some users to create
+            task_proxy = USERS.get_proxy(users_task)
+            log.debug("configuring users via %s task", task_proxy.Name)
+            sync_run_task(task_proxy)
+
+        root_task = users_proxy.SetRootPasswordWithTask((util.getSysroot()))
+        if self._root_password_already_configured and not reconfig_mode:
+            log.debug("skipping root password configuration - already configured")
+        else:
+            task_proxy = USERS.get_proxy(root_task)
+            log.debug("configuring root password via %s task", task_proxy.Name)
+            sync_run_task(task_proxy)
```

Comment:
Please move the tasks objects to the `else` parts. There is no need to create them if they are not used because the values are already configured.

Reply:
Good point, changed & thanks! :)